test/integration/admin_has_a_private_dashboard_test.rb

```diff
@@ -0,0 +1,75 @@
+require 'test_helper'
+
+class AdminHasAPrivateDashboardTest < ActionDispatch::IntegrationTest
+  test "admin login to their dashboard through contractor form" do
+    admin = create(:admin)
+    visit login_path
+    # save_and_open_page
+    within "#contractor-login-form" do
+      fill_in "session[email]", with: admin.email
+      fill_in "session[password]", with: admin.password
+      click_on "Contractor Login"
+    end
+    # save_and_open_page
+    assert_equal admin_dashboard_path, current_path
+    assert page.has_content? "Admin Dashboard"
+
+    visit root_path
+    visit '/admin/dashboard'
+    assert page.has_content? "Admin Dashboard"
+  end
+
+  test "admin login to their dashboard through developer form" do
+    admin = create(:admin)
+    visit login_path
+
+    within "#developer-login-form" do
+      fill_in "session[email]", with: admin.email
+      fill_in "session[password]", with: admin.password
+      click_on "Developer Login"
+    end
+
+    assert_equal admin_dashboard_path, current_path
+    assert page.has_content? "Admin Dashboard"
+
+    visit root_path
+    visit '/admin/dashboard'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.